lib/ansible/executor/task_executor.py

```diff
@@ -269,28 +269,39 @@ def _squash_items(self, items, variables):
             if all(isinstance(o, string_types) for o in items):
                 final_items = []
                 name = self._task.args.pop('name', None) or self._task.args.pop('pkg', None)
-                # The user is doing an upgrade or some other operation
-                # that doesn't take name or pkg.
-                if name:
+
+                # This gets the information to check whether the name field
+                # contains a template that we can squash for
+                template_no_item = template_with_item = None
+                if templar._contains_vars(name):
+                    variables['item'] = '\0$'
+                    template_no_item = templar.template(name, variables, cache=False)
+                    variables['item'] = '\0@'
+                    template_with_item = templar.template(name, variables, cache=False)
+                    del variables['item']
+
+                # Check if the user is doing some operation that doesn't take
+                # name/pkg or the name/pkg field doesn't have any variables
+                # and thus the items can't be squashed
+                if name and (template_no_item != template_with_item):
                     for item in items:
                         variables['item'] = item
                         if self._task.evaluate_conditional(templar, variables):
```

Comment:
It can return false but I should be restoring the value of name here: https://github.com/ansible/ansible/pull/14168/files#diff-76ffc0551f8cf3d6255500316568e60bR296  and returning hte items we were passed in unchanged here: https://github.com/ansible/ansible/pull/14168/files#diff-76ffc0551f8cf3d6255500316568e60bL294